src/hotspot/share/gc/serial/defNewGeneration.inline.hpp

```diff
@@ -91,11 +91,12 @@ inline void DefNewGeneration::FastKeepAliveClosure::do_oop_work(T* p) {
 
 template <typename OopClosureType>
 void DefNewGeneration::oop_since_save_marks_iterate(OopClosureType* cl) {
-  eden()->oop_since_save_marks_iterate(cl);
-  to()->oop_since_save_marks_iterate(cl);
-  from()->oop_since_save_marks_iterate(cl);
+  // No allocation in eden and from spaces, so no iteration required.
+  assert(eden()->saved_mark_at_top(), "inv");
+  assert(from()->saved_mark_at_top(), "inv");
```

Comment:
Instead of "inv", you could say "No allocation expected in {Eden|From}" (repsectivley).

Glad the assertions were added.

I can't imagine that this would make much of a difference in performance since the calls to eden and from's oop_since_save_marks_iterate would return immediately without doing any work, but I agree it's good to have the assertions as a check against unintended behavior.

Reply:
> Instead of "inv", you could say "No allocation expected in {Eden|From}" (repsectivley).

Went for the current approach to keep duplication to the minimal.